lib/ansible/modules/network/fortios/fortios_api.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/python
+#
+# Ansible module to manage arbitrary objects via API in fortigate devices
+# (c) 2017, Will Wagner <willwagner602@gmail.com> and Eugene Opredelennov <eoprede@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: fortios_api
+extends_documentation_fragment: fortios_api_doc
+version_added: "2.5"
+short_description: Manages arbitrarily designated fortios configurations.
+description:
+    - Manages arbitrarily designated fortios configurations. Can be used to edit any endpoint, as long as proper API path is provided.
+author:
+    - Will Wagner (@willwagner602)
+      Eugene Opredelennov (@eoprede)
+notes:
+    - Tested against Fortigate v5.4.5 VM
+
+options:
+    endpoint_information:
+      description: Information about the endpoint.
+      required: true
+      suboptions:
+            endpoint:
+                required: true
+                description: identifies the string to be added to the api
+                             URL to access the appropriate endpoint.
+            list_identifier:
+                required: true
+                description: The identifier used by this configuration to identify the list of
+                             objects that should be used to update the endpoint's configuration.
+            object_identifier:
+                required: false
+                description: The unique identifier used by this endpoint to differentiate objects.
+                             Typically 'name', 'id', or some variation. Not required for endpoints which
+                             contain only a single object.
+            permanent_objects:
+                required: false
+                description: A list of identifiers for objects at the endpoint that cannot be deleted.
+
+'''
+EXAMPLES = '''
+- name: Update router static
+  fortios_api:
+  endpoint_information:
```

Comment:
Indentation looks wrong here and in `conn_params`

Reply:
Thanks, that's fixed now.